Sources/ApplicativeRouter/Combinators.swift

```diff
@@ -0,0 +1,190 @@
+import Foundation
+import Prelude
+import Optics
+
+/// Processes and consumes a single path component matching the string provided.
+public func lit(_ str: String) -> Router<Prelude.Unit> {
```

Comment:
I put all the request routing specific combinators in its own file so it's easy to see what is core printer-parser versus routing specific.

Reply:
also almost all of these combinators are straight forward ports from what we had previously.